prime-router/src/main/kotlin/azure/DatabaseAccess.kt

```diff
@@ -650,6 +649,23 @@ class DatabaseAccess(private val create: DSLContext) : Logging {
             ) > 0
     }
 
+    /**
+     * Get the number of outstanding actions to batch for a specific receiver
+     */
+    fun fetchNumberOutstandingBatchRecords(
+        receiverFullName: String,
+        txn: DataAccessTransaction
+    ): Int {
+        val backstop = OffsetDateTime.now().minusDays(7)
+        return DSL.using(txn)
+            .select(TASK.asterisk())
+            .from(TASK)
+            .where(TASK.NEXT_ACTION.eq(TaskAction.batch))
```

Comment:
Do we need to also check that a batch action hasn't happened already? Or does it set the next action to none once batch has run for the task?

Reply:
It sets batch action to NONE once done.  If it fails, it will leave it as BATCH but an exception will be thrown, leading Jim (or whoever is monitoring) to look into it. This is intentional, but can lead to a death spiral if not rectified.